### PR TITLE
feat(tabs): implement overflow tabs and sync with v11

### PR DESCRIPTION
**Pull request by emyarod**
### Related Ticket(s)

#10086

### Description

This PR updates the tabs to match the upstream React version

Note: still missing icon-only variants

### Changelog

**New**

- implement overflow tabs

**Changed**

- update stories

**Removed**

- remove unused styles and breakpoints
- remove dropdown mobile tabs logic

### CODE REVIEW

**Comment by oliviaflory** [comment]
## Default tabs 
- [ ] Hover state: missing change on hover
   - text-primary
   - border-strong
   
Screenshot 2023-06-22 at 10 34 58 AM

---

- [ ] Focus state: missing style changes
   - $heading-compact-01
   - $text-primary

Screenshot 2023-06-22 at 10 33 08 AM

---

- [ ] Tab order differs from [Carbon React](https://react.carbondesignsystem.com/?path=/story/components-tabs--default).

| Web components | React |
| --- | --- |
| tab-order-wc |  tab-order-react |
| <ul><li>`tab` first tab item</li><li>`tab` second tab item</li><li>`tab` third tab item</li><li>`tab` fourth tab item</li><li> does not tab into any tab panel</li></ul> | <ul><li> `tab` first tab item</li><li>`tab` first tab item panel area</li></ul> |

## Contained tabs

- [ ] Hover state: missing change on hover
   - background: $layer-accent-hover  
   - text: $text-primary

Screenshot 2023-06-22 at 11 10 56 AM

---

- [ ] Tab order: I can't tab into the tab panel area

## Skeleton

- Looks different than React, but I think we can skip that for now and address later

## Playground

- [ ] Both contained and default (light variant) have an extended container, I'm not sure we need that?
Screenshot 2023-06-22 at 11 07 29 AM

---

- [ ] Theres a knob for a `light` variant of the default/line tabs, I don't think that exists in [v11](https://react.carbondesignsystem.com/?path=/story/components-tabs--playground) anymore? 
Screenshot 2023-06-22 at 11 09 41 AM

---

- [ ] **Tab type knob**: the `Container type (container)` doesn't seem to switch to the Contained style of tabs. Maybe we don't need that knob because the `Container type (contained)` works as expected?

Screenshot 2023-06-22 at 11 13 13 AM

**Comment by emyarod** [comment]
@oliviaflory for some reason GitHub isn't letting me check off some of the remaining checkboxes on your comment but I think I resolved everything except:

* skeleton states
* tabbing into tab panels

which I will implement in a future release!

**Comment by IgnacioBecerra** [comment]
@emyarod There is an error preventing the checks from completing:
https://github.com/carbon-design-system/carbon-for-ibm-dotcom/actions/runs/5382180256/jobs/9767238241?pr=10562#step:5:762

**Comment by emyarod** [comment]
thanks @IgnacioBecerra! I think I removed my original fixes for that when I was merging our branches

File: packages/carbon-web-components/src/components/tabs/tabs.ts [hunk 1]
```diff
@@ -180,105 +121,135 @@ class BXTabs extends HostListenerMixin(CDSContentSwitcher) {
...
+      default:
+        break;
+    }
+  }
+
+  @HostListener('scroll')
+  protected _handleScroll = () => {
+    debounce(this.requestUpdate as () => void, 500);
+    this.requestUpdate();
+  };
```
Comment (IgnacioBecerra) [inline packages/carbon-web-components/src/components/tabs/tabs.ts:204]: we shouldn't need this anymore

File: packages/carbon-web-components/src/components/tabs/tabs.ts [hunk 2]
```diff
@@ -180,105 +121,135 @@ class BXTabs extends HostListenerMixin(CDSContentSwitcher) {
...
+
+  _handleSlotchange() {
+    console.log('asdf');
```
Comment (IgnacioBecerra) [inline packages/carbon-web-components/src/components/tabs/tabs.ts:207]:
```suggestion
```

File: packages/carbon-web-components/src/components/tabs/tabs.ts [hunk 3]
```diff
@@ -305,37 +276,223 @@ class BXTabs extends HostListenerMixin(CDSContentSwitcher) {
...
+      }
+    }
+  }
+
+  disconnectedCallback() {
+    this._cleanAndCreateIntersectionObserverContainer();
+    this._contentContainerNode?.removeEventListener(
+      'scroll',
+      this._handleScroll
+    );
```
Comment (IgnacioBecerra) [inline packages/carbon-web-components/src/components/tabs/tabs.ts:373]:
```suggestion
```
can remove this too

**Review by IgnacioBecerra — COMMENTED** [review]
just removing these things but otherwise lgtm!

File: packages/carbon-web-components/src/components/tabs/tabs.scss [hunk 4]
```diff
@@ -84,51 +112,44 @@
...
+    color: $text-primary;
+  }
 }
 
-:host(#{$prefix}-tab[type='container']) {
-  @include breakpoint(md) {
-    background-color: $layer-accent-01;
+:host(#{$prefix}-tabs[type='contained'])
+  .#{$prefix}--tabs-nav-content-container {
+  height: rem(48px);
```
Comment (kennylam) [inline packages/carbon-web-components/src/components/tabs/tabs.scss:126]:
```suggestion
  height: rem($spacing-09);
```

File: packages/carbon-web-components/src/components/tabs/tabs.scss [hunk 5]
```diff
@@ -84,51 +112,44 @@
...
-    }
+:host(#{$prefix}-tab[type='contained']) {
+  background-color: $layer-accent-01;
 
-    + .#{$prefix}--tabs__nav-item.#{$prefix}--tabs__nav-item--selected,
-    &.#{$prefix}--tabs__nav-item--selected + .#{$prefix}--tabs__nav-item {
-      box-shadow: none;
-    }
+  a.#{$prefix}--tabs__nav-link {
+    height: rem(48px);
```
Comment (kennylam) [inline packages/carbon-web-components/src/components/tabs/tabs.scss:133]:
```suggestion
    height: rem($spacing-09);
```

File: packages/carbon-web-components/src/components/tabs/tabs.scss [hunk 6]
```diff
@@ -84,51 +112,44 @@
...
+    padding: $spacing-03 $spacing-05;
+    // height - vertical padding
+    /* stylelint-disable declaration-property-unit-whitelist */
+    line-height: calc(#{rem(48px)} - (#{$spacing-03} * 2));
```
Comment (kennylam) [inline packages/carbon-web-components/src/components/tabs/tabs.scss:137]:
```suggestion
    line-height: calc(#{$spacing-09} - (#{$spacing-03} * 2));
```

File: packages/carbon-web-components/src/components/tabs/tabs.scss [hunk 7]
```diff
@@ -180,65 +198,28 @@
...
+:host(#{$prefix}-tab[type='contained'][selected]:hover) {
+  background-color: $layer-01;
 
-    .#{$prefix}--tabs__nav-link:focus,
-    .#{$prefix}--tabs__nav-link:active {
-      box-shadow: none;
-    }
+  .#{$prefix}--tabs__nav-link {
+    // height - vertical padding
+    line-height: calc(#{rem(48px)} - (#{$spacing-03} * 2));
```
Comment (kennylam) [inline packages/carbon-web-components/src/components/tabs/tabs.scss:209]:
```suggestion
    line-height: calc(#{$spacing-09} - (#{$spacing-03} * 2));
```

**Review by kennylam — COMMENTED** [review]
Thanks @emyarod! Few small things.

**Review by kennylam — APPROVED** [review]
LGTM thanks @emyarod!

**Pull request merged**